LoRaEngine/LoraKeysManagerFacade/LoRaDevAddrCache.cs

```diff
@@ -238,10 +240,22 @@ private async Task<List<DevAddrCacheInfo>> GetDeviceTwinsFromIotHub(IDeviceRegis
                             NwkSKey = twin.GetNwkSKey(),
                             LastUpdatedTwins = twin.Properties.Desired.GetLastUpdated()
                         });
+
+                        if (!isFullReload
+                            && twin.Properties.Desired.GetMetadata() is { LastUpdated: { } desiredUpdateTime }
+                            && twin.Properties.Reported.GetMetadata() is { LastUpdated: { } reportedUpdateTime })
+                        {
+                            lastDeltaUpdateFromCache = new[] { lastDeltaUpdateFromCache, desiredUpdateTime.ToUniversalTime(), reportedUpdateTime.ToUniversalTime() }.Max();
```

Comment:
I assume the desiredUpdateTime and reportedUpdateTime have to be in UTC.
The problem with this approach is that you keep on allocating a new array for each iteration / device. If we have a few thousand of devices (the first load, without an entry in the lastDeltaUpdateFromCache and there was a bulk insert), this can be adding up. We could do something like:

```suggestion
lastDeltaUpdateFromCacheTicks  = Math.Max(lastDeltaUpdateFromCacheTicks, Math.Max(desiredUpdateTime.Ticks, reportedUpdateTime.Ticks))
```
You keep a new variable (last update in ticks) and in the end, you either store the ticks (preferred) or you convert it to a date: new Date(ticks)

Reply:
Using ticks with commit c502852f